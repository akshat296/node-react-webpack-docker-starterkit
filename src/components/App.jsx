/*
    ./client/components/App.jsx
*/
import React from 'react';
import { Card } from 'semantic-ui-react'
import { Header } from 'semantic-ui-react'

export default class App extends React.Component {
  render() {
    return (
      <div className="app">
        <Header as='h1'>React Semanti-UI Example</Header>
        <Card.Group>
          <Card>
            <Card.Content>
              <Card.Header>Matthew Harris</Card.Header>
              <Card.Meta>Co-Worker</Card.Meta>
              <Card.Description>Matthew is a pianist living in Nashville.</Card.Description>
            </Card.Content>
          </Card>

          <Card>
            <Card.Content>
              <Card.Header content='Jake Smith' />
              <Card.Meta content='Musicians' />
              <Card.Description content='Jake is a drummer living in New York.' />
            </Card.Content>
          </Card>

          <Card>
            <Card.Content
              header='Elliot Baker'
              meta='Friend'
              description='Elliot is a music producer living in Chicago.'
            />
          </Card>

          <Card
            header='Jenny Hess'
            meta='Friend'
            description='Jenny is a student studying Media Management at the New School'
          />
        </Card.Group>
      </div>

    )
  }
}
