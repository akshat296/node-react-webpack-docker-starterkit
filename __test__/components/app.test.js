import React from 'react';
import ReactDom from 'react-dom';
import App from '../../src/components/App'
import renderer from 'react-test-renderer';


/**
 * Testing the components using ReactDom
 */
describe("App", ()=> {
    it("renders the app without crashing", ()=>{
        const div = document.createElement("div");
        ReactDom.render(<App />, div);
    })
});

/**
 *  Tesing components using Snapshot testing using Jest
 */

 describe("Check whether App renders correctly usign snapshot", ()=>{
     it("renders correctly", () => {
         const rendered = renderer.create(
             <App />
         )
         expect(rendered.toJSON()).toMatchSnapshot();
     })
 })