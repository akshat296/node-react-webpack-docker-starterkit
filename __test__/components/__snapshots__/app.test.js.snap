// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check whether App renders correctly usign snapshot renders correctly 1`] = `
<div
  className="app"
>
  <h1
    className="ui header"
  >
    React Semanti-UI Example
  </h1>
  <div
    className="ui cards"
  >
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Matthew Harris
        </div>
        <div
          className="meta"
        >
          Co-Worker
        </div>
        <div
          className="description"
        >
          Matthew is a pianist living in Nashville.
        </div>
      </div>
    </div>
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Jake Smith
        </div>
        <div
          className="meta"
        >
          Musicians
        </div>
        <div
          className="description"
        >
          Jake is a drummer living in New York.
        </div>
      </div>
    </div>
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Elliot Baker
        </div>
        <div
          className="meta"
        >
          Friend
        </div>
        <div
          className="description"
        >
          Elliot is a music producer living in Chicago.
        </div>
      </div>
    </div>
    <div
      className="ui card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Jenny Hess
        </div>
        <div
          className="meta"
        >
          Friend
        </div>
        <div
          className="description"
        >
          Jenny is a student studying Media Management at the New School
        </div>
      </div>
    </div>
  </div>
</div>
`;
